### PR TITLE
Binary search in dart language

### LINKED ISSUE #2175: Binary Search
- [x] C
- [x] C++
- [x] C#
- [x] CoffeeScript
- [x] Dart - @keshakaneria
- [x] Go
- [x] Java
- [x] Julia
- [x] JS
- [x] Kotlin
- [x] Php
- [x] Python
- [x] Readme
- [x] Ruby
- [x] Rust - @raksha009
- [x] TypeScript - @Shr03mink

### PR DESCRIPTION
**Pull request by keshakaneria**
Fixes #2175 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Languages Used:
- Dart

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by keshakaneria** [comment]
@jainaman224 have a look at this code of binary search in dart language and suggest if any corrections

**Comment by raghav-dalmia** [comment]
Your PR has one irrelevant file, remove it and squash your commits.

**Comment by keshakaneria** [comment]
Deleted that file, named linear search @raghav-dalmia

**Comment by keshakaneria** [comment]
@raghav-dalmia code changed to dynamic values and solving other issues

**Comment by keshakaneria** [comment]
@raghav-dalmia any checks of this PR?

File: Binary_Search/Binary_Search.dart [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+                    print('Failed to insert element in List!');
+                    continue;
+                }
+                list.add(val);
+                list.sort();
+                print('Successfully inserted $val in List!');
+                break;
+
+            case 2:
+                int val = null;
```
Comment (singh-shreya6) [inline Binary_Search/Binary_Search.dart:66]: Hi, Delete functionality is not needed in this program. It is for doing binary search alone. Please remove the switch case for delete.

File: Binary_Search/Binary_Search.dart [hunk 2]
```diff
@@ -0,0 +1,175 @@
+import 'dart:io';
```
Comment (singh-shreya6) [inline Binary_Search/Binary_Search.dart:1]: Please add a brief problem description on top.

File: Binary_Search/Binary_Search.dart [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+                continue;
+        }
+    }
+}
+
+
+/*
+Sample Input/Output:
+Binary Search Program
+0 - Display List
```
Comment (singh-shreya6) [inline Binary_Search/Binary_Search.dart:116]: Remove the delete element case from output as well.

File: Binary_Search/Binary_Search.dart (hunk 1 above)
Comment (keshakaneria) [reply Binary_Search/Binary_Search.dart:66]: Deleted

**Comment by keshakaneria** [comment]
@singh-shreya6 As told, Delete functionality is removed. Kindly review it again.

**Comment by singh-shreya6** [comment]
Please squash your commits.

**Comment by keshakaneria** [comment]
@singh-shreya6 I have squashed the commits. Kindly now merge and evaluate me.

**Comment by singh-shreya6** [comment]
The commits are not squashed. The commit count should be 1 here instead of 8.

**Comment by singh-shreya6** [comment]
Please squash this one as well.

**Comment by keshakaneria** [comment]
@singh-shreya6 Squashed the commits

**Pull request merged**